tests/src/Kernel/HtmlPurifierFilterTest.php

```diff
@@ -40,10 +41,12 @@ public function testRemoveEmpty() {
     $processed = $this->filter->process($input, 'und')->getProcessedText();
     self::assertSame($expected, $processed);
 
-    // Set configuration to remove empty tags.
-    $config = \Drupal::service('config.factory')->getEditable('htmlpurifier.config_directives');
-    $config->set('AutoFormat.RemoveEmpty', TRUE)->save();
-    $this->filter->setDrupalConfig(\Drupal::config('htmlpurifier.config_directives')->get());
+    $configuration = [
+      'AutoFormat' => [
+        'RemoveEmpty' => TRUE,
+      ],
+    ];
+    $this->filter->settings['htmlpurifier_configuration'] = Yaml::encode($configuration);
```

Comment:
Woah, `\Drupal\filter\Plugin\FilterBase::$settings` is public?!?! That's such an oversight for core, but obviously makes this simpler here.